gui/default/syncthing/core/syncthingController.js

```diff
@@ -1047,6 +1047,96 @@ angular.module('syncthing.core')
             return deviceCfg.deviceID.substr(0, 6);
         };
 
+        $scope.downloadFile = function (httpPath) {
```

Comment:
I'm not sure what's going on here because I didn't read it. :) Can we not serve the zip file with the appropriate `Content-Disposition` header and let the browser be responsible for saving the file under the correct name etc?

Reply:
Yeah this is an overkill, you just need `<a href="/system/support" target="_blank>`

Reply:
@calmh @AudriusButkevicius 
I made some changes on code to get Content-Disposition header appropriately. Apart from that, I tried to cut off the unnecessary parts. Can you review my code?

        $scope.getSupportBundle = function (){
            $http.get(urlbase + '/system/support', { responseType: 'arraybuffer' })
             .success( function (data, status, headers) {
                var header = headers()["content-disposition"]
                var filename = header.split(';')[1].trim().split('=')[1];
                filename.replace(/"/g, '');
                var contentType = headers['content-type']
                // Get the blob url
                    var url = window.URL || window.webkitURL || window.mozURL || window.msURL;
                    if(url)
                    {
                        var a = document.createElement('a');
                        var blob = new Blob([data], { type: contentType });
                        var url = url.createObjectURL(blob);
                        a.setAttribute('href', url);
                        a.setAttribute("download", filename);
                        var event = document.createEvent('MouseEvents');
                        event.initMouseEvent('click', true, true, window, 1, 0, 0, 0, 0, false, false, false, false, 0, null);
                        a.dispatchEvent(event);
                    };
             });
            };

Reply:
@AudriusButkevicius thx for your quick reply. I try to use <a href="/system/support" target="_blank> but i get '404 page not found.' error. 
<img width="406" alt="code" src="https://user-images.githubusercontent.com/29825419/44633870-85855b80-a999-11e8-81e3-58ebd8ab196c.png">

Reply:
Because the url is prefixed with /rest/ etc, I just gave this as an ideological example.

Reply:
@AudriusButkevicius sorry to misunderstanding 'literally' word. I'll try it.

Reply:
Yeah, my bad, literally meant "all you need to do is just tell the browser to open the url".